libclamav_rust/src/scanners.rs

```diff
@@ -127,3 +134,164 @@ pub unsafe extern "C" fn scan_onenote(ctx: *mut cli_ctx) -> cl_error_t {
+        if header.is_directory() {
+            debug!("Skipping directory {filename}");
+        } else {
+            debug!("Found file in LHA archive: {filename}");
+
+            // Scan the archive metadata first.
+            if scan_archive_metadata(
+                ctx,
+                &filename,
+                header.compressed_size as usize,
+                header.original_size as usize,
+                false,
+                index,
+                header.file_crc as i32,
+            ) != cl_error_t_CL_SUCCESS
+            {
+                debug!("Extracted file '{filename}' would exceed size limits. Skipping.");
+            } else {
+                // Check if scanning the next file would exceed the limits and should be skipped.
+                if check_scan_limits("LHA", ctx, header.original_size, 0, 0)
+                    != cl_error_t_CL_SUCCESS
+                {
+                    debug!("Extracted file '{filename}' would exceed size limits. Skipping.");
+                } else {
+                    if !decoder.is_decoder_supported() {
+                        debug!("err: unsupported compression method");
+                    } else {
+                        // Read the file into a buffer.
+                        let mut file_data = Vec::<u8>::new();
+
+                        match decoder.read_to_end(&mut file_data) {
+                            Ok(bytes_read) => {
+                                if bytes_read > 0 {
+                                    let ret =
+                                        magic_scan(ctx, &file_data, Some(filename.to_string()));
+                                    if ret != cl_error_t_CL_SUCCESS {
+                                        debug!("cl_scandesc_magic returned error: {}", ret);
+                                        return ret;
+                                    }
+                                } else {
+                                    debug!("Read zero-byte file.");
+                                }
+                            }
+                            err => {
+                                debug!("Error reading file {err:?}");
+                            }
+                        }
+                    }
+                }
+            }
+
+            index += 1;
+        }
+
+        // Get the next file.
+        match decoder.next_file() {
+            Ok(true) => {
+                debug!("Found another file in the archive!");
+            }
+            Ok(false) => {
+                debug!("No more files in the archive.");
+                break;
+            }
+            Err(err) => {
+                // Error getting the next file.
+                // Use debug-level because may not actually be an LHA/LZH archive.
+                // LHA/LZH does not have particularly identifiable magic bytes.
+                debug!("An error occured when checking for the next file in this LHA or LZH archive: {err}");
```

Comment:
Typo, 'occurred'

Reply:
For the life of me I can never remember how to spell that.

Reply:
I only noticed because I was searching for it in the code, then had to google it.

Reply:
Fixed.